// xsolla-uikit/lib/_private/upsell/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardBannerWithoutDropdown = exports.CardBannerWithDropdown = undefined;

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

// $FlowFixMe - cant import 'useState' because there is no 'useState' export in 'react'


var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _xsollaPattern = require('../../_images/xsolla-pattern.svg');

var _xsollaPattern2 = _interopRequireDefault(_xsollaPattern);

var _index = require('../../index');

var _cardBanner = require('../../card-banner');

var _cardBanner2 = _interopRequireDefault(_cardBanner);

var _upsellFooter = require('./upsell-footer');

var _upsellFooter2 = _interopRequireDefault(_upsellFooter);

var _upsellChild = require('./upsell-child');

var _upsellChild2 = _interopRequireDefault(_upsellChild);

var _upsellDropdown = require('./upsell-dropdown');

var _upsellDropdown2 = _interopRequireDefault(_upsellDropdown);

var _exampleStyle = require('../../card-banner/exampleStyle.css');

var _exampleStyle2 = _interopRequireDefault(_exampleStyle);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind2.default.bind(_exampleStyle2.default);

var CardBannerWithDropdown = exports.CardBannerWithDropdown = function CardBannerWithDropdown() {
  var _useState = (0, _react.useState)(false),
      _useState2 = _slicedToArray(_useState, 2),
      isAssignToAM = _useState2[0],
      setAssignToAM = _useState2[1];

  var toggleFetching = function toggleFetching() {
    setTimeout(function () {
      setAssignToAM(function (prevState) {
        return {
          isAssignToAM: !prevState.isAssignToAM
        };
      });
    }, 1000);
  };

  return _react2.default.createElement(
    _react.Fragment,
    null,
    _react2.default.createElement(_cardBanner2.default, {
      className: cx('example-banner-card'),
      image: _xsollaPattern2.default,
      badge: 'Partner Network News',
      footer: _react2.default.createElement(_upsellFooter2.default, {
        components: _react2.default.createElement(_upsellDropdown2.default, {
          children: _react2.default.createElement(_index.Icon, { name: 'more', color: '#747e8a', hoverable: true }),
          onAssignToColleague: function onAssignToColleague() {
            return void 0;
          },
          onRequestHelp: toggleFetching,
          assignToAM: {
            disabled: isAssignToAM,
            label: 'Assign to Xsolla',
            description: 'description',
            sentDescription: 'was sent'
          },
          assignToColleague: {
            label: 'Assign to collegue',
            description: 'description',
            show: true
          },
          showLater: true,
          hideCardLabel: 'Not interested'
        }),
        submitLabel: 'Integrate',
        onSubmit: function onSubmit() {
          return void 0;
        }
      }),
      children: _react2.default.createElement(_upsellChild2.default, {
        title: 'Title',
        description: 'Description',
        caption: '5 min',
        moreLink: '/components/card-banner',
        onMoreLinkClick: function onMoreLinkClick() {
          return void 0;
        }
      })
    })
  );
};

var CardBannerWithoutDropdown = exports.CardBannerWithoutDropdown = function CardBannerWithoutDropdown() {
  return _react2.default.createElement(
    _react.Fragment,
    null,
    _react2.default.createElement(_cardBanner2.default, {
      className: cx('example-banner-card'),
      image: _xsollaPattern2.default,
      badge: 'Partner Network News',
      footer: _react2.default.createElement(_upsellFooter2.default, { submitLabel: 'Integrate', onSubmit: function onSubmit() {
          return void 0;
        } }),
      children: _react2.default.createElement(_upsellChild2.default, {
        title: 'Title',
        description: 'Description',
        caption: '5 min',
        moreLink: '/components/card-banner',
        onMoreLinkClick: function onMoreLinkClick() {
          return void 0;
        }
      })
    })
  );
};
