// xsolla-uikit/lib/_private/upsell/upsell-footer/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _index = require('../../../index');

var _style = require('./style.css');

var _style2 = _interopRequireDefault(_style);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind2.default.bind(_style2.default);

var UpsellFooter = function UpsellFooter(_ref) {
  var onSubmit = _ref.onSubmit,
      submitLabel = _ref.submitLabel,
      submitDataId = _ref.submitDataId,
      components = _ref.components,
      _ref$fetching = _ref.fetching,
      fetching = _ref$fetching === undefined ? false : _ref$fetching,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === undefined ? false : _ref$disabled;
  return _react2.default.createElement(
    'div',
    { className: cx('xsui-upsell-footer') },
    _react2.default.createElement(
      _index.Button,
      {
        onClick: onSubmit,
        size: 'sm',
        'data-id': submitDataId,
        fetching: fetching,
        disabled: disabled },
      submitLabel || 'Get started'
    ),
    components && components
  );
};

UpsellFooter.propTypes = {
  onSubmit: _propTypes2.default.func.isRequired,
  submitLabel: _propTypes2.default.string,
  fetching: _propTypes2.default.bool,
  disabled: _propTypes2.default.bool,
  submitDataId: _propTypes2.default.string,
  components: function components() {
    return (typeof _propTypes2.default.node === 'function' ? _propTypes2.default.node : _propTypes2.default.shape(_propTypes2.default.node)).apply(this, arguments);
  }
};
exports.default = UpsellFooter;
