// xsolla-uikit/lib/form-group/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ExampleTWP = function (_Component) {
  _inherits(ExampleTWP, _Component);

  function ExampleTWP() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, ExampleTWP);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = ExampleTWP.__proto__ || Object.getPrototypeOf(ExampleTWP)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      value: []
    }, _this.handleChange = function (value) {
      return _this.setState({ value: value });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(ExampleTWP, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _index2.default,
          { label: 'Primary' },
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto, pariatur?'
        ),
        _react2.default.createElement(
          _index2.default,
          { label: 'Primary with action', action: _react2.default.createElement(
              'span',
              null,
              'Action button'
            ), border: 'xs' },
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto, pariatur?'
        ),
        _react2.default.createElement(
          _index2.default,
          {
            label: 'Secondary with border & tooltip',
            appearance: 'secondary',
            indentation: 'lg',
            border: true,
            tooltip: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit' },
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda at atque, culpa laboriosam numquam voluptas!'
        ),
        _react2.default.createElement(
          _index2.default,
          {
            label: 'Secondary with tooltip',
            appearance: 'secondary',
            indentation: 'lg',
            tooltip: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit' },
          'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda at atque, culpa laboriosam numquam voluptas!'
        )
      );
    }
  }]);

  return ExampleTWP;
}(_react.Component);

exports.default = ExampleTWP;
