// xsolla-uikit/lib/input-group/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _input = require('../input');

var _input2 = _interopRequireDefault(_input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var InputGroupExample = function (_Component) {
  _inherits(InputGroupExample, _Component);

  function InputGroupExample() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, InputGroupExample);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = InputGroupExample.__proto__ || Object.getPrototypeOf(InputGroupExample)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      minValue: 1,
      maxValue: 15
    }, _this.setMinValue = function (value) {
      return _this.setState({ minValue: value });
    }, _this.setMaxValue = function (value) {
      return _this.setState({ maxValue: value });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(InputGroupExample, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        _formGroup2.default,
        { label: 'Separate' },
        _react2.default.createElement(
          _index2.default,
          { appearance: 'separate' },
          _react2.default.createElement(
            'div',
            { style: { width: '128px' } },
            _react2.default.createElement(_input2.default, {
              input: {
                value: this.state.minValue,
                onChange: function onChange(e) {
                  return _this2.setMinValue(e.target.value);
                }
              },
              disabledLimitationMinWidth: true
            })
          ),
          _react2.default.createElement(
            'div',
            { style: { width: '128px' } },
            _react2.default.createElement(_input2.default, {
              input: {
                value: this.state.maxValue,
                onChange: function onChange(e) {
                  return _this2.setMaxValue(e.target.value);
                }
              },
              disabledLimitationMinWidth: true
            })
          )
        )
      );
    }
  }]);

  return InputGroupExample;
}(_react.Component);

exports.default = InputGroupExample;
