// xsolla-uikit/lib/file-picker/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Example = function (_PureComponent) {
  _inherits(Example, _PureComponent);

  function Example() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Example);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Example.__proto__ || Object.getPrototypeOf(Example)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      value: ''
    }, _this.setValue = function (e) {
      return _this.setState({ value: e.target.value });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Example, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { style: { maxWidth: 480 } },
        _react2.default.createElement(
          _formGroup2.default,
          {
            label: '\u0421\u0442\u0430\u043D\u0434\u0430\u0440\u0442\u043D\u044B\u0439 \u0437\u0430\u0433\u0440\u0443\u0437\u0447\u0438\u043A',
            caption: this.state.value && 'URL: ' + this.state.value,
            indentation: 'lg' },
          _react2.default.createElement(_index2.default, {
            input: { name: 'default', value: this.state.value, onChange: this.setValue },
            requestConfig: { url: 'http://xsolla.com/', field: 'file' },
            sizeLimit: 5 * 1000000
          })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: '\u041E\u0448\u0438\u0431\u043A\u0430 \u043F\u0440\u0438 \u0437\u0430\u0433\u0440\u0443\u0437\u043A\u0435', indentation: 'lg' },
          _react2.default.createElement(_index2.default, {
            input: { name: 'default', value: 'image.png', onChange: function onChange() {
                return void 0;
              } },
            meta: {
              touched: true,
              error: 'Uploading failed. Check file requirements and try again.'
            },
            requestConfig: { url: 'http://xsolla.com/', field: 'file' }
          })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: '\u041D\u0435\u0430\u043A\u0442\u0438\u0432\u043D\u044B\u0439 \u0437\u0430\u0433\u0440\u0443\u0437\u0447\u0438\u043A', indentation: 'lg' },
          _react2.default.createElement(_index2.default, {
            input: { name: 'default', value: this.state.value, onChange: this.setValue },
            requestConfig: { url: 'http://xsolla.com/', field: 'file' },
            disabled: true
          })
        )
      );
    }
  }]);

  return Example;
}(_react.PureComponent);

exports.default = Example;
