// xsolla-uikit/lib/head/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _button = require('../button');

var _button2 = _interopRequireDefault(_button);

var _icon = require('../icon');

var _icon2 = _interopRequireDefault(_icon);

var _dropdown = require('../dropdown');

var _dropdown2 = _interopRequireDefault(_dropdown);

var _contentBlock = require('../content-block');

var _contentBlock2 = _interopRequireDefault(_contentBlock);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function () {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Primary', indentation: 'lg' },
      _react2.default.createElement(
        _contentBlock2.default,
        null,
        _react2.default.createElement(
          _index2.default,
          null,
          'Head content'
        ),
        _react2.default.createElement(
          'p',
          null,
          'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Aut eius dolorum pariatur! Eligendi blanditiis qui, quis assumenda sapiente fugiat, illo tenetur velit iste doloribus voluptatum.'
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Primary, centered', indentation: 'lg' },
      _react2.default.createElement(
        _contentBlock2.default,
        null,
        _react2.default.createElement(
          _index2.default,
          { align: 'center' },
          'Head content'
        ),
        _react2.default.createElement(
          'p',
          null,
          'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Aut eius dolorum pariatur! Eligendi blanditiis qui, quis assumenda sapiente fugiat, illo tenetur velit iste doloribus voluptatum.'
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Separated', indentation: 'lg' },
      _react2.default.createElement(
        _contentBlock2.default,
        null,
        _react2.default.createElement(
          _index2.default,
          { appearance: 'separated' },
          'Head content'
        ),
        _react2.default.createElement(
          'p',
          null,
          'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Aut eius dolorum pariatur! Eligendi blanditiis qui, quis assumenda sapiente fugiat, illo tenetur velit iste doloribus voluptatum.'
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Extra', indentation: 'lg' },
      _react2.default.createElement(
        _contentBlock2.default,
        null,
        _react2.default.createElement(
          _index2.default,
          { appearance: 'separated', extra: _react2.default.createElement(
              _button2.default,
              { appearance: 'flatten' },
              'Extra button'
            ) },
          'Head content'
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Separated with \'xlg\' indentation & extra dropdown', indentation: 'lg' },
      _react2.default.createElement(
        _contentBlock2.default,
        null,
        _react2.default.createElement(
          _index2.default,
          {
            appearance: 'separated',
            indentation: 'xlg',
            extra: _react2.default.createElement(
              _dropdown2.default,
              {
                size: 'sm',
                align: 'right',
                autosize: true,
                options: [{
                  label: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.\n              Fugiat labore exercitationem eligendi modi consectetur quos.',
                  icon: 'hint-filled'
                }, { label: 'Analytics', icon: 'analytics' }, { label: 'Update', description: 'Update this item', icon: 'update' }] },
              _react2.default.createElement(
                _button2.default,
                { appearance: 'flatten' },
                _react2.default.createElement(_icon2.default, { name: 'expand' }),
                ' Flatten trigger'
              )
            ) },
          'Head test content'
        )
      )
    )
  );
};
