// xsolla-uikit/lib/head/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _style = require('./style.css');

var _style2 = _interopRequireDefault(_style);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var cx = _bind2.default.bind(_style2.default);

var Head = function Head(_ref) {
  var children = _ref.children,
      _ref$appearance = _ref.appearance,
      appearance = _ref$appearance === undefined ? 'primary' : _ref$appearance,
      _ref$indentation = _ref.indentation,
      indentation = _ref$indentation === undefined ? 'md' : _ref$indentation,
      _ref$align = _ref.align,
      align = _ref$align === undefined ? 'left' : _ref$align,
      extra = _ref.extra;
  return React.createElement(
    'div',
    {
      className: cx('xsui-head', 'xsui-head--appearance-' + appearance, 'xsui-head--indentation-' + indentation, 'xsui-head--align-' + align) },
    React.createElement(
      'div',
      { className: cx('xsui-head__title') },
      ' ',
      children
    ),
    extra && React.createElement(
      'div',
      { className: cx('xsui-head__extra') },
      extra
    )
  );
};

Head.propTypes = {
  children: _propTypes2.default.node.isRequired,
  appearance: _propTypes2.default.oneOf(['primary', 'separated']),
  indentation: _propTypes2.default.oneOf(['none', 'sm', 'md', 'lg', 'xlg']),
  align: _propTypes2.default.oneOf(['left', 'center']),
  extra: _propTypes2.default.node
};
exports.default = Head;
