// xsolla-uikit/lib/dropdown/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _button = require('../button');

var _button2 = _interopRequireDefault(_button);

var _icon = require('../icon');

var _icon2 = _interopRequireDefault(_icon);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _pill = require('../pill');

var _pill2 = _interopRequireDefault(_pill);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// @todo move to another component

var Row = function Row(_ref) {
  var children = _ref.children;
  return _react2.default.createElement(
    'div',
    { className: 'example__row' },
    children
  );
};

Row.propTypes = {
  children: _propTypes2.default.any.isRequired
};

exports.default = function () {
  return _react2.default.createElement(
    _react.Fragment,
    null,
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Large dropdown' },
      _react2.default.createElement(
        _index2.default,
        {
          size: 'lg',
          autosize: true,
          options: [{
            label: 'Lorem ipsum',
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.\n              Fugiat labore exercitationem eligendi modi consectetur quos.',
            icon: 'hint-filled'
          }, { label: 'Analytics', icon: 'analytics' }, { label: _react2.default.createElement(
              'em',
              null,
              'Update'
            ), description: _react2.default.createElement(
              'strong',
              null,
              'Update this item'
            ), icon: 'update' }] },
        _react2.default.createElement(
          _button2.default,
          { appearance: 'flatten' },
          _react2.default.createElement(_icon2.default, { name: 'expand' }),
          ' Flatten trigger'
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Any element' },
      _react2.default.createElement(
        'div',
        { style: { display: 'flex', alignItems: 'center' } },
        _react2.default.createElement(
          _index2.default,
          {
            align: 'left',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(_icon2.default, { name: 'more' })
        ),
        '\xA0 \xA0 \xA0',
        _react2.default.createElement(
          _index2.default,
          {
            align: 'left',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(
            _button2.default,
            { appearance: 'square' },
            _react2.default.createElement(_icon2.default, { name: 'hint-filled' })
          )
        ),
        '\xA0 \xA0 \xA0',
        _react2.default.createElement(
          _index2.default,
          {
            align: 'left',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(
            _button2.default,
            { appearance: 'secondary' },
            'Trigger'
          )
        )
      )
    ),
    _react2.default.createElement(
      'div',
      { style: { display: 'flex', width: '660px' } },
      _react2.default.createElement(
        _formGroup2.default,
        { label: 'Icon align: \'right\'' },
        _react2.default.createElement(
          _index2.default,
          {
            align: 'right',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(_icon2.default, { name: 'more' })
        )
      ),
      _react2.default.createElement(
        _formGroup2.default,
        { label: 'Icon align: \'left\'' },
        _react2.default.createElement(
          _index2.default,
          {
            align: 'left',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(_icon2.default, { name: 'more' })
        )
      ),
      _react2.default.createElement(
        _formGroup2.default,
        { label: 'Icon align: \'left-half\'' },
        _react2.default.createElement(
          _index2.default,
          {
            align: 'left-half',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(_icon2.default, { name: 'more' })
        )
      ),
      _react2.default.createElement(
        _formGroup2.default,
        { label: 'Icon align: \'left-half-back\'' },
        _react2.default.createElement(
          _index2.default,
          {
            align: 'left-half-back',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(_icon2.default, { name: 'more' })
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Wide options' },
      _react2.default.createElement(
        _index2.default,
        {
          size: 'lg',
          autosize: true,
          options: [{
            label: 'Lorem ipsum',
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.\n                Fugiat labore exercitationem eligendi modi consectetur quos.',
            icon: 'hint-filled'
          }, { label: 'Analytics', icon: 'analytics' }, { label: 'Update', description: 'Update this item', icon: 'update' }],
          optionsWidth: 'lg' },
        _react2.default.createElement(
          _button2.default,
          { appearance: 'flatten' },
          _react2.default.createElement(_icon2.default, { name: 'expand' }),
          ' Flatten trigger'
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Information: right' },
      _react2.default.createElement(
        _index2.default,
        {
          size: 'lg',
          type: 'inform',
          align: 'right',
          autosize: true,
          separated: false,
          content: {
            title: 'Lorem ipsum',
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.\n            Fugiat labore exercitationem eligendi modi consectetur quos.',
            button: 'Connect',
            buttonAppearance: 'secondary'
          } },
        _react2.default.createElement(
          _pill2.default,
          { appearance: 'cancelled', separated: true },
          'Flatten trigger'
        )
      )
    ),
    _react2.default.createElement(
      'div',
      { style: { display: 'flex', width: '600px' } },
      _react2.default.createElement(
        _formGroup2.default,
        { label: 'Open onHover' },
        _react2.default.createElement(
          _index2.default,
          {
            trigger: 'hover',
            autosize: true,
            separated: false,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(_icon2.default, { name: 'more' })
        )
      )
    ),
    _react2.default.createElement(
      'div',
      { style: { display: 'flex', width: '600px' } },
      _react2.default.createElement(
        _formGroup2.default,
        { label: 'Disabled' },
        _react2.default.createElement(
          _index2.default,
          {
            disabled: true,
            options: [{ label: 'Settings', icon: 'settings' }, { label: 'Delete', color: '#eb002f', icon: 'delete' }] },
          _react2.default.createElement(
            _button2.default,
            { appearance: 'flatten' },
            _react2.default.createElement(_icon2.default, { name: 'expand' }),
            ' Disabled trigger'
          )
        )
      )
    )
  );
};
