// xsolla-uikit/lib/button-action/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTransitionGroup = require('react-transition-group');

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Example = function (_Component) {
  _inherits(Example, _Component);

  function Example() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Example);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Example.__proto__ || Object.getPrototypeOf(Example)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      items: [new Date().valueOf()],
      customItems: ['Custom value']
    }, _this.pushItem = function () {
      return _this.setState({ items: [].concat(_toConsumableArray(_this.state.items), [new Date().valueOf()]) });
    }, _this.pushCustomItem = function () {
      return _this.setState({ customItems: [].concat(_toConsumableArray(_this.state.customItems), ['']) });
    }, _this.removeItem = function (index) {
      return _this.setState({ items: _this.state.items.filter(function (a, key) {
          return key !== index;
        }) });
    }, _this.removeCustomItem = function (index) {
      return _this.setState({ customItems: _this.state.customItems.filter(function (a, key) {
          return key !== index;
        }) });
    }, _this.changeValue = function (index, newValue) {
      return _this.setState({
        customItems: _this.state.customItems.map(function (a, key) {
          return key === index ? newValue : a;
        })
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Example, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        _react.Fragment,
        null,
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Primary', style: { width: 468 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'primary', icon: 'add-circle', onClick: this.pushItem },
            'Create item'
          )
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Filled with extra label', style: { width: 468 } },
          _react2.default.createElement(
            _reactTransitionGroup.TransitionGroup,
            null,
            this.state.items.map(function (item, index) {
              return _react2.default.createElement(
                _reactTransitionGroup.CSSTransition,
                { key: 'item--' + index, timeout: 100, classNames: 'slide-fade-44' },
                _react2.default.createElement(
                  _formGroup2.default,
                  { indentation: 'sm' },
                  _react2.default.createElement(
                    _index2.default,
                    {
                      appearance: 'filled',
                      extraLabel: '$3.99',
                      icon: 'delete',
                      onClick: _this2.removeItem.bind(_this2, index) },
                    item
                  )
                )
              );
            })
          )
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'With custom text', style: { width: 468 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'primary', icon: 'add-circle', onClick: this.pushCustomItem },
            'Create item'
          )
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Input with custom text', style: { width: 468 } },
          _react2.default.createElement(
            _reactTransitionGroup.TransitionGroup,
            null,
            this.state.customItems.map(function (item, index) {
              return _react2.default.createElement(
                _reactTransitionGroup.CSSTransition,
                { key: 'custom-item--' + index, timeout: 100, classNames: 'slide-fade-44' },
                _react2.default.createElement(
                  _formGroup2.default,
                  { indentation: 'sm' },
                  _react2.default.createElement(
                    _index2.default,
                    {
                      appearance: 'input',
                      icon: 'delete',
                      placeholder: 'Custom placeholder',
                      onClick: _this2.removeCustomItem.bind(_this2, index),
                      onChange: _this2.changeValue.bind(_this2, index) },
                    item
                  )
                )
              );
            })
          )
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Disabled', style: { width: 468 } },
          _react2.default.createElement(
            _formGroup2.default,
            { indentation: 'sm' },
            _react2.default.createElement(
              _index2.default,
              {
                appearance: 'filled',
                extraLabel: '$3.99',
                icon: 'delete',
                disabled: true,
                onClick: function onClick() {
                  return void 0;
                } },
              'Disabled button'
            )
          )
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Error', style: { width: 468 } },
          _react2.default.createElement(
            _index2.default,
            {
              meta: { touched: true, error: 'You need to add at least 1 item.' },
              icon: 'add-circle' },
            'Create item'
          )
        )
      );
    }
  }]);

  return Example;
}(_react.Component);

exports.default = Example;
