// xsolla-uikit/lib/card-banner/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _example = require('../_private/upsell/example');

var _xsollaPattern = require('../_images/xsolla-pattern.svg');

var _xsollaPattern2 = _interopRequireDefault(_xsollaPattern);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _button = require('../button');

var _button2 = _interopRequireDefault(_button);

var _index3 = require('../index');

var _exampleStyle = require('./exampleStyle.css');

var _exampleStyle2 = _interopRequireDefault(_exampleStyle);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind2.default.bind(_exampleStyle2.default);

var ExampleBannerCard = function ExampleBannerCard() {
  return _react2.default.createElement(_index2.default, {
    className: cx('example-banner-card'),
    image: _xsollaPattern2.default,
    badge: 'Partner Network News',
    footer: _react2.default.createElement(
      _index3.FormFooter,
      { ignoreCentered: true },
      _react2.default.createElement(
        _button2.default,
        null,
        'Connect'
      )
    ),
    children: _react2.default.createElement(
      _formGroup2.default,
      { indentation: 'sm' },
      'Partner Network lets you manage every affiliate and influencer program with a single, easy-to-use, cost-effective tool.'
    )
  });
};

exports.default = function () {
  return _react2.default.createElement(
    'div',
    { style: { backgroundColor: '#f5f9ff', margin: '-40px -60px', padding: '40px 60px' } },
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Product card', indentation: 'lg' },
      _react2.default.createElement(ExampleBannerCard, null)
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Product card with upsell child and footer' },
      _react2.default.createElement(
        'div',
        { className: cx('upsell-cards-container') },
        _react2.default.createElement(_example.CardBannerWithoutDropdown, null),
        _react2.default.createElement(_example.CardBannerWithDropdown, null)
      )
    )
  );
};
