// xsolla-uikit/lib/date-picker/classNames.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _style = require('./style.css');

var _style2 = _interopRequireDefault(_style);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind2.default.bind(_style2.default);

exports.default = {
  container: cx('xsui-datepicker'),
  wrapper: cx('xsui-datepicker__wrapper'),
  interactionDisabled: cx('xsui-datepicker--disabled'),
  months: cx('xsui-datepicker__months'),
  month: cx('xsui-datepicker__month'),

  navBar: cx('xsui-datepicker__nav-bar'),
  navButtonPrev: cx('xsui-datepicker__nav-bar__button xsui-datepicker__nav-bar__button--prev'),
  navButtonNext: cx('xsui-datepicker__nav-bar__button xsui-datepicker__nav-bar__button--next'),
  navButtonInteractionDisabled: cx('xsui-datepicker__nav-bar__button--disabled'),

  caption: cx('xsui-datepicker__caption'),
  weekdays: cx('xsui-datepicker__weekdays'),
  weekdaysRow: cx('xsui-datepicker__weekdays__row'),
  weekday: cx('xsui-datepicker__weekdays__item'),
  body: cx('xsui-datepicker__body'),
  week: cx('xsui-datepicker__week'),
  weekNumber: cx('xsui-datepicker__week-number'),
  day: cx('xsui-datepicker__day'),
  footer: cx('xsui-datepicker__footer'),
  todayButton: cx('xsui-datepicker__today-button'),

  start: cx('xsui-datepicker__day--start'),
  startCompleted: cx('xsui-datepicker__day--start-completed'),
  end: cx('xsui-datepicker__day--end'),
  endHovered: cx('xsui-datepicker__day--end-hovered'),
  today: cx('xsui-datepicker__day--today'),
  selected: cx('xsui-datepicker__day--selected'),
  disabled: cx('xsui-datepicker__day--disabled'),
  outside: cx('xsui-datepicker__day--outside')
};
