// xsolla-uikit/lib/pill/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _dropdown = require('../dropdown');

var _dropdown2 = _interopRequireDefault(_dropdown);

var _statusDot = require('../status-dot');

var _statusDot2 = _interopRequireDefault(_statusDot);

var _icon = require('../icon');

var _icon2 = _interopRequireDefault(_icon);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var appearances = ['active', 'branding', 'warning', 'error', 'info', 'cancelled'];

exports.default = function () {
  return _react2.default.createElement(
    _react.Fragment,
    null,
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'No action', indentation: 'xxlg' },
      _react2.default.createElement(
        'div',
        { style: { display: 'flex' } },
        _react2.default.createElement(
          'div',
          { style: { marginRight: 32, maxWidth: 140 } },
          _react2.default.createElement(
            _index2.default,
            null,
            'Completed Completed CompletedCompletedCompletedCompleted'
          )
        ),
        _react2.default.createElement(
          'div',
          { style: { marginRight: 32 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'branding' },
            'Open'
          )
        ),
        _react2.default.createElement(
          'div',
          { style: { marginRight: 32 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'warning' },
            'Attention'
          )
        ),
        _react2.default.createElement(
          'div',
          { style: { marginRight: 32 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'error' },
            'Error'
          )
        ),
        _react2.default.createElement(
          'div',
          { style: { marginRight: 32 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'info' },
            'Testing'
          )
        ),
        _react2.default.createElement(
          'div',
          { style: { marginRight: 32 } },
          _react2.default.createElement(
            _index2.default,
            { appearance: 'cancelled' },
            'Cancelled'
          )
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Redirection', indentation: 'xxlg' },
      _react2.default.createElement(
        'a',
        { href: 'https://uikit.xsolla.com/components/pill' },
        _react2.default.createElement(
          _index2.default,
          { appearance: 'warning', clickable: true },
          _react2.default.createElement(
            'span',
            null,
            'Test webhooks'
          ),
          ' ',
          _react2.default.createElement(_icon2.default, { name: 'chevron-mini' })
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Selector', indentation: 'xxlg' },
      _react2.default.createElement(
        _formGroup2.default,
        { indentation: 'xs' },
        _react2.default.createElement(
          'div',
          { style: { display: 'flex' } },
          appearances.map(function (appearance, index) {
            return _react2.default.createElement(
              'div',
              { key: 'appearance--' + index, style: { marginRight: 32 } },
              _react2.default.createElement(
                _dropdown2.default,
                {
                  autosize: true,
                  separated: true,
                  options: [{
                    label: 'Make as Master',
                    description: 'Build will replace current Master',
                    customIcon: _react2.default.createElement(_statusDot2.default, { status: 'enabled' }),
                    iconAlign: 'right'
                  }, {
                    label: 'Make as Test',
                    description: 'Build will not be available for some devs',
                    customIcon: _react2.default.createElement(_statusDot2.default, { status: 'testing' }),
                    iconAlign: 'right'
                  }] },
                _react2.default.createElement(
                  _index2.default,
                  { appearance: appearance },
                  _react2.default.createElement(_icon2.default, { name: 'settings' })
                )
              )
            );
          })
        )
      ),
      _react2.default.createElement(
        _formGroup2.default,
        { indentation: 'sm' },
        _react2.default.createElement(
          'div',
          { style: { display: 'flex' } },
          appearances.map(function (appearance, index) {
            return _react2.default.createElement(
              'div',
              { key: 'appearance--' + index, style: { marginRight: 32 } },
              _react2.default.createElement(
                _dropdown2.default,
                {
                  autosize: true,
                  separated: true,
                  options: [{
                    label: 'Make as Master',
                    description: 'Build will replace current Master',
                    customIcon: _react2.default.createElement(_statusDot2.default, { status: 'enabled' }),
                    iconAlign: 'right'
                  }, {
                    label: 'Make as Test',
                    description: 'Build will not be available for some devs',
                    customIcon: _react2.default.createElement(_statusDot2.default, { status: 'testing' }),
                    iconAlign: 'right'
                  }] },
                _react2.default.createElement(
                  _index2.default,
                  { appearance: appearance, fetching: true },
                  _react2.default.createElement(_icon2.default, { name: 'settings' })
                )
              )
            );
          })
        )
      ),
      _react2.default.createElement(
        _formGroup2.default,
        { indentation: 'sm' },
        _react2.default.createElement(
          'div',
          { style: { display: 'flex' } },
          appearances.map(function (appearance, index) {
            return _react2.default.createElement(
              'div',
              { key: 'appearance--' + index, style: { marginRight: 32 } },
              _react2.default.createElement(
                _dropdown2.default,
                {
                  autosize: true,
                  separated: true,
                  options: [{
                    label: 'Make as Master',
                    description: 'Build will replace current Master',
                    customIcon: _react2.default.createElement(_statusDot2.default, { status: 'enabled' }),
                    iconAlign: 'right'
                  }, {
                    label: 'Make as Test',
                    description: 'Build will not be available for some devs',
                    customIcon: _react2.default.createElement(_statusDot2.default, { status: 'testing' }),
                    iconAlign: 'right'
                  }] },
                _react2.default.createElement(
                  _index2.default,
                  { appearance: appearance, separated: true },
                  appearance
                )
              )
            );
          })
        )
      )
    )
  );
};
