// xsolla-uikit/lib/checkbox-group/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var options = [{ id: 1, label: 'Troopers', extraLabel: '$4.00' }, { id: 3, label: 'Planes' }, { id: 4, label: 'Tanks' }, {
  id: 5,
  label: 'Sharks',
  options: [{ id: 7, label: 'Catshark' }, { id: 8, label: 'Angelshark' }, { id: 9, label: 'Wobbegong' }]
}, { id: 6, label: 'Disabled item', disabled: true }];

var ExampleCheckboxGroup = function (_Component) {
  _inherits(ExampleCheckboxGroup, _Component);

  function ExampleCheckboxGroup() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, ExampleCheckboxGroup);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = ExampleCheckboxGroup.__proto__ || Object.getPrototypeOf(ExampleCheckboxGroup)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      multiple: [1],
      empty: []
    }, _this.push = function (item) {
      setTimeout(function () {
        _this.setState({ multiple: [].concat(_toConsumableArray(_this.state.multiple), [item]) });
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(ExampleCheckboxGroup, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        'div',
        { style: { width: 468 } },
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'With stripes' },
          _react2.default.createElement(_index2.default, {
            name: 'basic',
            fields: {
              getAll: function getAll() {
                return _this2.state.multiple;
              },
              removeAll: function removeAll() {
                return _this2.setState({ multiple: [] });
              },
              push: this.push,
              remove: function remove(index) {
                var arr = [].concat(_toConsumableArray(_this2.state.multiple));
                arr.splice(index, 1);
                _this2.setState({
                  multiple: arr
                });
              }
            },
            options: options
          })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Without stripes and allow select nested folder' },
          _react2.default.createElement(_index2.default, {
            name: 'basic',
            showStripes: false,
            allowSelectNestedFolder: true,
            fields: {
              getAll: function getAll() {
                return _this2.state.multiple;
              },
              removeAll: function removeAll() {
                return _this2.setState({ multiple: [] });
              },
              push: this.push,
              remove: function remove(index) {
                var arr = [].concat(_toConsumableArray(_this2.state.multiple));
                arr.splice(index, 1);
                _this2.setState({
                  multiple: arr
                });
              }
            },
            options: options
          })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Disabled' },
          _react2.default.createElement(_index2.default, {
            name: 'basic',
            disabled: true,
            fields: {
              getAll: function getAll() {
                return _this2.state.multiple;
              },
              removeAll: function removeAll() {
                return _this2.setState({ multiple: [] });
              },
              push: this.push,
              remove: function remove(index) {
                var arr = [].concat(_toConsumableArray(_this2.state.multiple));
                arr.splice(index, 1);
                _this2.setState({
                  multiple: arr
                });
              }
            },
            options: options
          })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'With placeholder' },
          _react2.default.createElement(_index2.default, {
            name: 'basic',
            placeholder: 'No options',
            fields: {
              getAll: function getAll() {
                return _this2.state.empty;
              },
              removeAll: function removeAll() {
                return _this2.setState({ empty: [] });
              },
              push: function push(item) {
                return _this2.setState({ empty: [].concat(_toConsumableArray(_this2.state.empty), [item]) });
              },
              remove: function remove(index) {
                var arr = [].concat(_toConsumableArray(_this2.state.empty));
                arr.splice(index, 1);
                _this2.setState({
                  empty: arr
                });
              }
            },
            options: []
          })
        )
      );
    }
  }]);

  return ExampleCheckboxGroup;
}(_react.Component);

exports.default = ExampleCheckboxGroup;
