// xsolla-uikit/lib/collapse/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _input = require('../input');

var _input2 = _interopRequireDefault(_input);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var localesList = ['English', 'Russian', 'French'];
var bigText = 'Portal\u2122 is a new single player game from Valve. Set in the mysterious Aperture\nScience Laboratories, Portal has been called one of the most innovative new games on the horizon\nand will offer gamers hours of unique gameplay. The game is designed to change the way players\napproach, manipulate, and surmise the possibilities in a given environment; similar to how\nHalf-Life\xAE 2\'s Gravity Gun innovated new ways to leverage an object in any given situation.\nPlayers must solve physical puzzles and challenges by opening portals to maneuvering objects,\nand themselves, through space.';

exports.default = function () {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Basic', border: true },
      _react2.default.createElement(
        'div',
        { className: 'example__col', style: { width: 470 } },
        _react2.default.createElement(
          _index2.default,
          {
            isOpened: true,
            staticElements: 1,
            collapsedLabel: function collapsedLabel(count) {
              return count + ' more languages';
            },
            expandedLabel: function expandedLabel() {
              return 'Hide localizations';
            } },
          localesList.map(function (locale, index) {
            return _react2.default.createElement(
              _formGroup2.default,
              {
                key: index,
                label: 'Project name | ' + locale,
                indentation: 'sm',
                tooltip: 'Project name' },
              _react2.default.createElement(_input2.default, { input: { name: 'name.' + locale, value: null, onChange: function onChange() {
                    return void 0;
                  } } })
            );
          })
        )
      )
    ),
    _react2.default.createElement(
      _formGroup2.default,
      { label: 'Show elemnts inline' },
      _react2.default.createElement(
        'div',
        { className: 'example__col', style: { width: 470 } },
        _react2.default.createElement(
          _index2.default,
          {
            btnIconPosition: 'end',
            staticElements: 1,
            showElementsInline: true,
            collapsedLabel: function collapsedLabel() {
              return 'Show more';
            },
            expandedLabel: function expandedLabel() {
              return 'Collapse';
            } },
          [_react2.default.createElement(
            'span',
            { key: 0 },
            bigText.substr(0, 200)
          ), _react2.default.createElement(
            'span',
            { key: 1 },
            bigText.substr(200)
          )]
        )
      )
    )
  );
};
