// xsolla-uikit/lib/input-copy/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Example = function (_PureComponent) {
  _inherits(Example, _PureComponent);

  function Example() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, Example);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Example.__proto__ || Object.getPrototypeOf(Example)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      value: 'Some text'
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Example, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        { style: { maxWidth: 480, margin: '0 auto' } },
        _react2.default.createElement(
          _formGroup2.default,
          {
            label: 'Default',
            indentation: 'lg',
            caption: 'Copies input value to clipboard by icon click' },
          _react2.default.createElement(_index2.default, { input: { name: 'copy', value: this.state.value } })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          {
            label: 'With text selection',
            indentation: 'lg',
            caption: 'Selects field text (you can press copy keys manually)' },
          _react2.default.createElement(_index2.default, { input: { name: 'copy', value: this.state.value }, selectOnClick: true })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          {
            label: 'With onCopy event callback',
            indentation: 'lg',
            caption: 'Copies value and fires onCopy callback (shown in console)' },
          _react2.default.createElement(_index2.default, {
            input: { name: 'copy', value: this.state.value },
            onCopy: function onCopy() {
              console.log('onCopy callback'); // eslint-disable-line
            }
          })
        )
      );
    }
  }]);

  return Example;
}(_react.PureComponent);

exports.default = Example;
