// xsolla-uikit/lib/slider/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SliderExample = function (_Component) {
  _inherits(SliderExample, _Component);

  function SliderExample() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, SliderExample);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = SliderExample.__proto__ || Object.getPrototypeOf(SliderExample)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      value: 4,
      rangeValue: [4, 8, 12, 16]
    }, _this.setValue = function (value) {
      return _this.setState({ value: value });
    }, _this.setRangeValue = function (value) {
      return _this.setState({ rangeValue: value });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(SliderExample, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var props = {
        input: { name: 'slider', value: this.state.value, onChange: function onChange(value) {
            return _this2.setValue(value);
          } },
        meta: {
          touched: false,
          error: null
        },
        min: 0,
        max: 10,
        disabled: false,
        step: 0.1
      };

      var rangeProps = {
        input: {
          name: 'rangeSlider',
          value: this.state.rangeValue,
          onChange: function onChange(value) {
            return _this2.setRangeValue(value);
          }
        },
        meta: {
          touched: false,
          error: null
        },
        min: 0,
        max: 100,
        range: true,
        count: 3,
        disabled: false,
        step: 1
      };

      return _react2.default.createElement(
        _react.Fragment,
        null,
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Slider', indentation: 'lg' },
          _react2.default.createElement(_index2.default, props),
          _react2.default.createElement(
            'div',
            null,
            this.state.value
          )
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Ranged slider', indentation: 'lg' },
          _react2.default.createElement(_index2.default, rangeProps),
          _react2.default.createElement(
            'div',
            null,
            '[' + this.state.rangeValue[0] + ', ' + this.state.rangeValue[1] + ', ' + this.state.rangeValue[2] + ', ' + this.state.rangeValue[3] + ']'
          )
        )
      );
    }
  }]);

  return SliderExample;
}(_react.Component);

exports.default = SliderExample;
