// xsolla-uikit/lib/pagination/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var PaginationExample = function (_PureComponent) {
  _inherits(PaginationExample, _PureComponent);

  function PaginationExample() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, PaginationExample);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = PaginationExample.__proto__ || Object.getPrototypeOf(PaginationExample)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      currentPage: 1
    }, _this.handeChangePage = function (currentPage) {
      return _this.setState({ currentPage: currentPage });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(PaginationExample, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Small' },
          _react2.default.createElement(_index2.default, {
            size: 'sm',
            current: this.state.currentPage,
            total: 16,
            perPage: 1,
            onChangePage: this.handeChangePage
          })
        ),
        _react2.default.createElement(
          _formGroup2.default,
          { label: 'Medium' },
          _react2.default.createElement(_index2.default, {
            current: this.state.currentPage,
            total: 16,
            perPage: 1,
            renderItem: function renderItem(props) {
              return _react2.default.createElement('a', props);
            },
            onChangePage: this.handeChangePage
          })
        )
      );
    }
  }]);

  return PaginationExample;
}(_react.PureComponent);

exports.default = PaginationExample;
