// xsolla-uikit/lib/group-select/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _formGroup = require('../form-group');

var _formGroup2 = _interopRequireDefault(_formGroup);

var _countryOptions = require('../../utils/example-data/country-options');

var _countryOptions2 = _interopRequireDefault(_countryOptions);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var options = [{
  label: 'Africa',
  value: 'africa',
  children: [{ label: 'Algeria', value: 'algeria' }, { label: 'Angola', value: 'algeria1' }, { label: 'Benin', value: 'algeria2' }, { label: 'Perm', value: 'algeria3' }]
}, {
  label: 'Asia',
  value: 'asia',
  children: [{ label: 'Angola', value: 'angola' }, { label: 'Algeria', value: 'algeria' }, { label: 'Benin', value: 'benin' }, { label: 'Perm', value: 'perm' }]
}];

var optionsForAll = [{
  label: 'Africa',
  value: 'africa',
  children: [{ label: 'Angola', value: 'angola' }, { label: 'Algeria', value: 'algeria' }, { label: 'Kenya', value: 'kenya' }, { label: 'Rwanda', value: 'rwanda' }, { label: 'Kongo', value: 'kongo' }]
}, {
  label: 'Asia',
  value: 'asia',
  children: [{ label: 'Bangladesh', value: 'bangladesh' }, { label: 'Bhutan', value: 'bhutan' }, { label: 'Nepal', value: 'nepal' }, { label: 'Pakistan', value: 'pakistan' }, { label: 'Sri Lanka', value: 'srilanka' }]
}, {
  label: 'Europe',
  value: 'europe',
  children: [{ label: 'Russia', value: 'russia' }, { label: 'Germany', value: 'germany' }, { label: 'Italy', value: 'italy' }, { label: 'Spain', value: 'spain' }, { label: 'Austria', value: 'austria' }]
}];

var ExampleTWP = function (_Component) {
  _inherits(ExampleTWP, _Component);

  function ExampleTWP() {
    _classCallCheck(this, ExampleTWP);

    return _possibleConstructorReturn(this, (ExampleTWP.__proto__ || Object.getPrototypeOf(ExampleTWP)).apply(this, arguments));
  }

  _createClass(ExampleTWP, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _react.Fragment,
        null,
        _react2.default.createElement(
          WithExampleState,
          null,
          function (_ref) {
            var value = _ref.value,
                handleChange = _ref.handleChange;
            return _react2.default.createElement(
              _react.Fragment,
              null,
              _react2.default.createElement(
                'div',
                { className: 'example__row' },
                _react2.default.createElement(
                  'div',
                  { className: 'example__col', style: { width: 470 } },
                  _react2.default.createElement(
                    _formGroup2.default,
                    { label: 'Basic' },
                    _react2.default.createElement(_index2.default, {
                      name: 'basic',
                      input: { value: value, onChange: handleChange },
                      options: options
                    })
                  )
                )
              ),
              _react2.default.createElement(
                'div',
                { className: 'example__row' },
                _react2.default.createElement(
                  'div',
                  { className: 'example__col', style: { width: 470 } },
                  _react2.default.createElement(
                    _formGroup2.default,
                    { label: 'Disabled XS' },
                    _react2.default.createElement(_index2.default, {
                      name: 'basic-disabled',
                      input: { value: value, onChange: handleChange },
                      options: options,
                      disabled: true
                    })
                  )
                )
              )
            );
          }
        ),
        _react2.default.createElement(
          'div',
          { className: 'example__row' },
          _react2.default.createElement(
            'div',
            { className: 'example__col', style: { width: 470 } },
            _react2.default.createElement(
              _formGroup2.default,
              { label: 'Show group with all elements' },
              _react2.default.createElement(
                WithExampleState,
                null,
                function (_ref2) {
                  var value = _ref2.value,
                      handleChange = _ref2.handleChange;
                  return _react2.default.createElement(_index2.default, {
                    name: 'countries',
                    input: { value: value, onChange: handleChange },
                    options: optionsForAll,
                    showAllGroup: true,
                    allGroupLabelForInput: 'Worldwide',
                    allGroupLabel: 'All Regions'
                  });
                }
              )
            )
          )
        ),
        _react2.default.createElement(
          'div',
          { className: 'example__row' },
          _react2.default.createElement(
            'div',
            { className: 'example__col', style: { width: 470 } },
            _react2.default.createElement(
              _formGroup2.default,
              { label: 'With many values' },
              _react2.default.createElement(
                WithExampleState,
                null,
                function (_ref3) {
                  var value = _ref3.value,
                      handleChange = _ref3.handleChange;
                  return _react2.default.createElement(_index2.default, {
                    name: 'countries',
                    input: { value: value, onChange: handleChange },
                    options: _countryOptions2.default
                  });
                }
              )
            )
          )
        )
      );
    }
  }]);

  return ExampleTWP;
}(_react.Component);

var WithExampleState = function (_Component2) {
  _inherits(WithExampleState, _Component2);

  function WithExampleState() {
    var _ref4;

    var _temp, _this2, _ret;

    _classCallCheck(this, WithExampleState);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this2 = _possibleConstructorReturn(this, (_ref4 = WithExampleState.__proto__ || Object.getPrototypeOf(WithExampleState)).call.apply(_ref4, [this].concat(args))), _this2), _this2.state = {
      value: []
    }, _temp), _possibleConstructorReturn(_this2, _ret);
  }

  _createClass(WithExampleState, [{
    key: 'render',
    value: function render() {
      var _this3 = this;

      // eslint-disable-next-line react/prop-types
      return this.props.children(_extends({}, this.props, {
        value: this.state.value,
        handleChange: function handleChange(value) {
          return _this3.setState({ value: value });
        }
      }));
    }
  }]);

  return WithExampleState;
}(_react.Component);

exports.default = ExampleTWP;
