// xsolla-uikit/lib/_private/upsell/integration-guide/types.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bpfrpt_proptype_IntegrationGuideCardType = exports.bpfrpt_proptype_StepsType = exports.bpfrpt_proptype_StepType = exports.bpfrpt_proptype_StepTypeEnum = undefined;

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var bpfrpt_proptype_StepTypeEnum = {
  DEFAULT: _propTypes2.default.oneOf(['default']).isRequired,
  CONTACT_XSOLLA: _propTypes2.default.oneOf(['contact_xsolla']).isRequired
};
var bpfrpt_proptype_StepType = {
  id: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string]).isRequired,
  title: _propTypes2.default.string.isRequired,
  description: _propTypes2.default.string,
  url: _propTypes2.default.string,
  isExternal: _propTypes2.default.bool,
  contactLocked: _propTypes2.default.bool.isRequired,
  type: _propTypes2.default.oneOfType([_propTypes2.default.oneOf(['default']), _propTypes2.default.oneOf(['contact_xsolla'])]).isRequired
};

var bpfrpt_proptype_StepsType = _propTypes2.default.arrayOf(_propTypes2.default.shape({
  id: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string]).isRequired,
  title: _propTypes2.default.string.isRequired,
  description: _propTypes2.default.string,
  url: _propTypes2.default.string,
  isExternal: _propTypes2.default.bool,
  contactLocked: _propTypes2.default.bool.isRequired,
  type: _propTypes2.default.oneOfType([_propTypes2.default.oneOf(['default']), _propTypes2.default.oneOf(['contact_xsolla'])]).isRequired
}).isRequired);

var bpfrpt_proptype_IntegrationGuideCardType = {
  id: _propTypes2.default.number.isRequired,
  imageUrl: _propTypes2.default.string.isRequired,
  title: _propTypes2.default.string.isRequired,
  description: _propTypes2.default.string.isRequired,
  activeStepId: _propTypes2.default.number,
  steps: _propTypes2.default.arrayOf(_propTypes2.default.shape({
    id: _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.string]).isRequired,
    title: _propTypes2.default.string.isRequired,
    description: _propTypes2.default.string,
    url: _propTypes2.default.string,
    isExternal: _propTypes2.default.bool,
    contactLocked: _propTypes2.default.bool.isRequired,
    type: _propTypes2.default.oneOfType([_propTypes2.default.oneOf(['default']), _propTypes2.default.oneOf(['contact_xsolla'])]).isRequired
  }).isRequired).isRequired
};
exports.bpfrpt_proptype_StepTypeEnum = bpfrpt_proptype_StepTypeEnum;
exports.bpfrpt_proptype_StepType = bpfrpt_proptype_StepType;
exports.bpfrpt_proptype_StepsType = bpfrpt_proptype_StepsType;
exports.bpfrpt_proptype_IntegrationGuideCardType = bpfrpt_proptype_IntegrationGuideCardType;
