// xsolla-uikit/lib/image-picker/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

var _row = require('./row');

var _row2 = _interopRequireDefault(_row);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ExampleImagePicker = function (_PureComponent) {
  _inherits(ExampleImagePicker, _PureComponent);

  function ExampleImagePicker() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, ExampleImagePicker);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = ExampleImagePicker.__proto__ || Object.getPrototypeOf(ExampleImagePicker)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
      value: 'https://source.unsplash.com/random'
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(ExampleImagePicker, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      return _react2.default.createElement(
        _react.Fragment,
        null,
        _react2.default.createElement(
          _row2.default,
          null,
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'Empty'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'With image'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image2',
              input: { value: this.state.value, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'With error'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image3',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              meta: { touched: true, error: 'Error' },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'Disabled'
            ),
            _react2.default.createElement(_index2.default, {
              disabled: true,
              name: 'image',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          )
        ),
        _react2.default.createElement(
          _row2.default,
          null,
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'Empty'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image',
              appearance: 'tertiary',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'With image'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image2',
              appearance: 'tertiary',
              input: { value: this.state.value, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'With error'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image3',
              appearance: 'tertiary',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              meta: { touched: true, error: 'Error' },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'Disabled'
            ),
            _react2.default.createElement(_index2.default, {
              disabled: true,
              name: 'image',
              appearance: 'tertiary',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          )
        ),
        _react2.default.createElement(
          _row2.default,
          null,
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'Empty'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image',
              appearance: 'secondary',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'With image'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image2',
              appearance: 'secondary',
              input: { value: this.state.value, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'With error'
            ),
            _react2.default.createElement(_index2.default, {
              name: 'image3',
              appearance: 'secondary',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              meta: { touched: true, error: 'Error' },
              url: '/'
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(
              'h4',
              null,
              'Disabled'
            ),
            _react2.default.createElement(_index2.default, {
              disabled: true,
              name: 'image',
              appearance: 'secondary',
              input: { value: null, onChange: function onChange() {
                  return _this2.setState({ value: null });
                } },
              url: '/'
            })
          )
        )
      );
    }
  }]);

  return ExampleImagePicker;
}(_react.PureComponent);

exports.default = ExampleImagePicker;
