// xsolla-uikit/lib/checkbox/example.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Row = function Row(_ref) {
  var children = _ref.children;
  return _react2.default.createElement(
    'div',
    { className: 'example__row' },
    children
  );
};

Row.propTypes = {
  children: _propTypes2.default.node
};

var CheckboxExample = function (_PureComponent) {
  _inherits(CheckboxExample, _PureComponent);

  function CheckboxExample() {
    var _ref2;

    var _temp, _this, _ret;

    _classCallCheck(this, CheckboxExample);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref2 = CheckboxExample.__proto__ || Object.getPrototypeOf(CheckboxExample)).call.apply(_ref2, [this].concat(args))), _this), _this.state = {
      value: false
    }, _this.toggleValue = function () {
      return _this.setState({ value: !_this.state.value });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(CheckboxExample, [{
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        _react.Fragment,
        null,
        _react2.default.createElement(
          Row,
          { title: 'primary' },
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(_index2.default, {
              name: 'checkbox',
              label: 'Basic checkbox',
              input: {
                value: this.state.value,
                onChange: this.toggleValue
              }
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(_index2.default, {
              name: 'checkbox',
              label: 'Disabled checkbox',
              input: {
                value: this.state.value,
                onChange: this.toggleValue
              },
              disabled: true
            })
          ),
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(_index2.default, {
              name: 'checkbox',
              label: _react2.default.createElement(
                'span',
                { style: { fontWeight: 600 } },
                'Checkbox with tooltip and custom label'
              ),
              input: {
                value: this.state.value,
                onChange: this.toggleValue
              },
              tooltip: 'Tooltip message'
            })
          )
        ),
        _react2.default.createElement(
          Row,
          { title: 'with error' },
          _react2.default.createElement(
            'div',
            { className: 'example__col' },
            _react2.default.createElement(_index2.default, {
              name: 'checkbox',
              label: 'Checkbox',
              meta: { touched: true, error: 'Error' },
              input: {
                value: this.state.value,
                onChange: this.toggleValue
              }
            })
          )
        )
      );
    }
  }]);

  return CheckboxExample;
}(_react.PureComponent);

exports.default = CheckboxExample;
