// xsolla-uikit/lib/_private/upsell/upsell-child/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bind = require('classnames/bind');

var _bind2 = _interopRequireDefault(_bind);

var _style = require('./style.css');

var _style2 = _interopRequireDefault(_style);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var cx = _bind2.default.bind(_style2.default);
var PREFIX = 'xsui-upsell-entry';

var UpsellChild = function UpsellChild(_ref) {
  var title = _ref.title,
      description = _ref.description,
      moreLink = _ref.moreLink,
      caption = _ref.caption,
      onMoreLinkClick = _ref.onMoreLinkClick,
      titleDataId = _ref.titleDataId,
      descriptionDataId = _ref.descriptionDataId,
      captionDataId = _ref.captionDataId,
      learnMoreDataId = _ref.learnMoreDataId;
  return _react2.default.createElement(
    'div',
    { className: cx(PREFIX) },
    _react2.default.createElement(
      'div',
      { className: cx(PREFIX + '__title'), 'data-id': titleDataId },
      title
    ),
    _react2.default.createElement(
      'div',
      {
        className: cx(PREFIX + '__content', PREFIX + '__content--description', (moreLink || caption) && PREFIX + '__content--indentation-sm'),
        'data-id': descriptionDataId },
      description
    ),
    moreLink && _react2.default.createElement(
      'a',
      { href: moreLink, onClick: onMoreLinkClick, 'data-id': learnMoreDataId, target: '_blank' },
      'Learn more'
    ),
    caption && _react2.default.createElement(
      'div',
      {
        className: cx(PREFIX + '__content', PREFIX + '__content--caption'),
        'data-id': captionDataId },
      caption
    )
  );
};

UpsellChild.propTypes = {
  title: _propTypes2.default.string.isRequired,
  description: _propTypes2.default.string.isRequired,
  moreLink: _propTypes2.default.string,
  caption: _propTypes2.default.string,
  onMoreLinkClick: _propTypes2.default.func,
  titleDataId: _propTypes2.default.string,
  descriptionDataId: _propTypes2.default.string,
  captionDataId: _propTypes2.default.string,
  learnMoreDataId: _propTypes2.default.string
};
exports.default = UpsellChild;
